presto-main/src/main/java/io/prestosql/sql/ExpressionUtils.java

```diff
@@ -330,21 +327,6 @@ else if (!seen.contains(expression)) {
         return result.build();
     }
 
-    public static Expression normalize(Expression expression)
```

Comment:
Removing this piece is not only a cleanup. It has the status of a fix, as NaN wasn't handled correctly here. That could be mentioned in commit message.

Reply:
Good catch! I pulled it out as a separate PR: https://github.com/prestosql/presto/pull/4120